consensus/core/src/network/tonic_network.rs

```diff
@@ -604,21 +604,28 @@ impl<S: NetworkService> NetworkManager<S> for TonicManager {
                         }
                     }
                 } else {
-                    // Create TcpListener via TCP socket, for more fine grained configurations.
-                    let socket = if own_address.is_ipv4() {
-                        tokio::net::TcpSocket::new_v4()
-                    } else if own_address.is_ipv6() {
-                        tokio::net::TcpSocket::new_v6()
-                    } else {
-                        panic!("Invalid own address: {own_address:?}");
-                    }
-                    .unwrap_or_else(|e| panic!("Cannot create TCP socket: {e:?}"));
-                    if let Err(e) = socket.set_nodelay(true) {
-                        info!("Failed to set TCP_NODELAY: {e:?}");
-                    }
-                    if let Err(e) = socket.set_reuseaddr(true) {
-                        info!("Failed to set SO_REUSEADDR: {e:?}");
+                    let tcp_connection_metrics = &self.context.metrics.network_metrics.tcp_connection_metrics;
+
+                    // Try creating an ephemeral port to test the highest allowed send and recv buffer sizes.
+                    // Buffer sizes are not set explicitly on the socket used for real traffic,
+                    // to allow the OS to set appropriate values.
+                    {
+                        let ephemeral_addr = SocketAddr::new(own_address.ip(), 0);
+                        let ephemeral_socket = create_socket(&ephemeral_addr);
+                        if let Err(e) = ephemeral_socket.set_send_buffer_size(32 << 20) {
```

Comment:
Why 32 MB? Based [on our guide](https://github.com/MystenLabs/sui/blob/main/docs/content/guides/operator/validator-tasks.mdx#option-1-with-etcsysctld) we want the max to be ~100 MB